### PR TITLE
Added Nth-cdr function, and corresponding test

### PR DESCRIPTION
**Pull request by Izaakwltn**
Added nth-cdr, modeled after the index definition.

(Please ignore the aforementioned "'intercalatexf' typo change",  that was a personal emacs keyboard snafu)

### CODE REVIEW

File: library/list.lisp [hunk 1]
```diff
@@ -234,6 +235,16 @@
     "Like INDEX, but errors if the index is not found."
     (from-some "There is no NTH" (index n l)))
 
+  (declare nth-cdr (Ufix -> List :a -> (Optional (List :a))))
```
Comment (stylewarning) [inline library/list.lisp:238]: To be consistent with CDR this ought to just do CDR at most N times, shorting out when it reaches the empty list. 

Alternatively, to be consistent with NTH, we could error on (nth-cdr n Nil) for n>0. I don't recommend it though.

**Review by stylewarning — CHANGES_REQUESTED** [review]

**Comment by Izaakwltn** [comment]
I ran into an error using 'decleration' so I found and fixed all uses of 'decleration' instead of 'declaration' in the codebase

File: library/list.lisp [hunk 2]
```diff
@@ -234,6 +235,15 @@
     "Like INDEX, but errors if the index is not found."
     (from-some "There is no NTH" (index n l)))
 
+  (declare nth-cdr (Ufix -> List :a -> List :a))
+  (define (nth-cdr n l)
+    "Returns the nth-cdr of a list."
+    (cond ((null? l)
```
Comment (stylewarning) [inline library/list.lisp:241]: style: put condition on new line

File: library/list.lisp [hunk 3]
```diff
@@ -234,6 +235,15 @@
...
+	   nil)
```
Comment (stylewarning) [inline library/list.lisp:242]: style: capitalize Nil

File: library/list.lisp [hunk 4]
```diff
@@ -234,6 +235,15 @@
...
+	  ((== 0 n)
```
Comment (stylewarning) [inline library/list.lisp:243]: style: use `zero?`

File: library/list.lisp [hunk 5]
```diff
@@ -234,6 +235,15 @@
...
+	   l)
+	  (True (nth-cdr (- n 1) (cdr l)))))
```
Comment (stylewarning) [inline library/list.lisp:245]: style: use `1-`

**Review by stylewarning — APPROVED** [review]
only minor style things

in general the typo fixes should remain in a separate commit. will you be rebasing your commits to clean the history?

**Pull request closed without merging**